tree/30_generic_methods/file_ensure_lines_present.cf

```diff
@@ -30,13 +30,19 @@
 bundle agent file_ensure_lines_present(file, lines)
 {
   vars:
+      "old_class_prefix"  string => canonify("file_ensure_lines_present_${file}");
       "promisers"          slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
       "class_prefix"      string => canonify(join("_", "promisers"));
-      "args"              slist => { "${file}", "${lines}" };
+      "args"               slist => { "${file}", "${lines}" };
+      "canonified_file"   string => canonify("${file}");
+
 
   methods:
       "enforce lines content" usebundle => file_enforce_content("${file}", "${lines}", "false");
+      "new result classes"    usebundle => _classes_copy("${class_prefix}_action", "${class_prefix}"),
+                             ifvarclass => "${class_prefix}_action_reached";
+
       "new result classes"    usebundle => _classes_copy("${class_prefix}_enforce_lines_content", "${class_prefix}");
 
-      "report"                usebundle => _log("Insert content ${lines} into ${file}", "", "${class_prefix}", @{args});
+      "report"                usebundle => _log("Insert content ${lines} into ${file}", "${old_class_prefix}", "${class_prefix}", @{args});
```

Comment:
I fear this will make a double report

Reply:
Ok, this works as this makes the exact same report twice, can you remove this `${old_class_prefix}` to avoid confusion about who is making the actual report?